openapi/openapi/src/main/scala/endpoints4s/ujson/JsonSchemas.scala

```diff
@@ -29,7 +29,8 @@ trait JsonSchemas extends algebra.NoDocsJsonSchemas with TuplesSchemas {
 
   }
 
-  abstract class Tagged[A](val discriminator: String) extends Record[A] {
+  trait Tagged[A] extends Record[A] {
+    def discriminator: String
```

Comment:
Was there a reason for changing the parameter into an abstract method? (I’m fine with this change, which I believe is equivalent to the previous state, I’m just curious if there is something that _required_ the change or not)

Reply:
This is because in `lazyTagged` we return a `Tagged` instance as follow:

```scala
override def lazyTagged[A](name: String)(schema: => Tagged[A]): Tagged[A] = {
  lazy val evaluatedSchema = schema
  new Tagged[A] {
    // Here we need to keep the discriminator lazy
    lazy val discriminator = evaluatedSchema.discriminator
    def findDecoder(tag: String): Option[Decoder[ujson.Value, A]] =
      evaluatedSchema.findDecoder(tag)
    def tagAndObj(a: A): (String, ujson.Obj) =
      evaluatedSchema.tagAndObj(a)
  }
}
```